### PR TITLE
PR: dev_statistics --> main (#5) 

### PR DESCRIPTION
**Pull request by TildeLundqvist**
Files stats.h and stats.cc containing functions that calculates mean, median, mode, variance, and standard deviation.
File test_stats.cc containing tests testing the functionality on several cases including edge case: empty vector.
Corresponding BUILD files created and documentation added.

### CODE REVIEW

File: libs/math/stats.cc [hunk 1]
```diff
@@ -0,0 +1,177 @@
+#include "stats.h"
+
+/**
+ * Gets the scores from the data and converts them to integers.
+ * @param data the data
+ * @return if data is not empty, returns a vector of the scores, otherwise returns an empty vector
+ * @author Tilde
+ */
+vector<int> getScores(vector<vector<pair<string, string>>> data)
```
Comment (SamuelPalmhager) [inline libs/math/stats.cc:9]: Calculations look correct, however using const & for the parameters is best practice.

File: libs/math/stats.cc [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    return scores;
+}
+
+/**
+ * Calculates the mean of the scores.
+ * @param data the data
+ * @return if data is not empty, returns the mean of the scores, otherwise -1
+ * @author Tilde
+ */
+double Stats::calcMean(vector<vector<pair<string, string>>> data)
```
Comment (SamuelPalmhager) [inline libs/math/stats.cc:42]: Same as above, const &

File: libs/math/stats.cc [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    return sum/scores.size();
+}
+
+/**
+ * Calculates the median of the scores.
+ * @param data the data
+ * @return if data is not empty, returns the median of the scores, otherwise -1
+ * @author Tilde
+ */
+double Stats::calcMedian(vector<vector<pair<string, string>>> data)
```
Comment (SamuelPalmhager) [inline libs/math/stats.cc:66]: Same as above, const &

File: libs/math/stats.cc [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    return scores[middleIndex];
+}
+
+/**
+ * Calculates the mode of the scores.
+ * @param data the data
+ * @return if data is not empty, returns a vector with the modes of the scores, otherwise returns an empty vector
+ * @author Tilde
+ */
+vector<int> Stats::calcMode(vector<vector<pair<string, string>>> data)
```
Comment (SamuelPalmhager) [inline libs/math/stats.cc:93]: Same as above, const &

File: libs/math/stats.cc [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    return modes;
+}
+
+/**
+ * Calculates the variance of the scores.
+ * @param data the data
+ * @return if data is not empty, returns the variance of the scores, otherwise returns -1
+ * @author Tilde
+ */
+double Stats::calcVariance(vector<vector<pair<string, string>>> data)
```
Comment (SamuelPalmhager) [inline libs/math/stats.cc:134]: Same as above, const &

File: libs/math/stats.cc [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    return sum/scores.size();
+}
+
+/**
+ * Calculates the standard deviation of the scores.
+ * @param data the data
+ * @return if data is not empty, returns the standard deviation of the scores, otherwise returns -1
+ * @author Tilde
+ */
+double Stats::calcStandardDeviation(vector<vector<pair<string, string>>> data)
```
Comment (SamuelPalmhager) [inline libs/math/stats.cc:167]: Same as above, const &

File: tests/cpp/test_stats.cc [hunk 7]
```diff
@@ -0,0 +1,124 @@
+#include "libs/math/stats.h"
+#include <gtest/gtest.h>
+
+TEST(StatsTest, Mean) {
+    Stats stats;
+
+    vector<vector<pair<string,string>>> data = { { {"id", "1"}, {"name", "Alice"}, {"age", "23"}, {"score", "88"} } };
```
Comment (SamuelPalmhager) [inline tests/cpp/test_stats.cc:7]: Could use auto (?), makes code look cleaner.

File: tests/cpp/test_stats.cc [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    EXPECT_DOUBLE_EQ(stats.calcMean(data), 84.25);
+
+    data = { {} };
+    EXPECT_DOUBLE_EQ(stats.calcMean(data), -1.0);
+}
+
+TEST(StatsTest, Median) {
+    Stats stats;
+
+    vector<vector<pair<string,string>>> data = { { {"id", "1"}, {"name", "Alice"}, {"age", "23"}, {"score", "88"} } };
```
Comment (SamuelPalmhager) [inline tests/cpp/test_stats.cc:26]: Could use auto (?), makes code look cleaner.

File: tests/cpp/test_stats.cc [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    EXPECT_DOUBLE_EQ(stats.calcMedian(data), 85.0);
+
+    data = { {} };
+    EXPECT_DOUBLE_EQ(stats.calcMedian(data), -1.0);
+}
+
+TEST(StatsTest, Mode) {
+    Stats stats;
+
+    vector<vector<pair<string,string>>> data = { { {"id", "1"}, {"name", "Alice"}, {"age", "23"}, {"score", "88"} } };
```
Comment (SamuelPalmhager) [inline tests/cpp/test_stats.cc:45]: Could use auto (?), makes code look cleaner.

File: tests/cpp/test_stats.cc [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+    data = { {} };
+    result = stats.calcMode(data);
+    ASSERT_TRUE(result.empty());
+}
+
+TEST(StatsTest, Variance) {
+    Stats stats;
+
+    vector<vector<pair<string,string>>> data = { { {"id", "1"}, {"name", "Alice"}, {"age", "23"}, {"score", "88"} } };
```
Comment (SamuelPalmhager) [inline tests/cpp/test_stats.cc:91]: Could use auto (?), makes code look cleaner.

File: tests/cpp/test_stats.cc [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    EXPECT_DOUBLE_EQ(stats.calcVariance(data), 26.1875);
+
+    data = { {} };
+    EXPECT_DOUBLE_EQ(stats.calcVariance(data), -1.0);
+}
+
+TEST(StatsTest, StandardDeviation) {
+    Stats stats;
+
+    vector<vector<pair<string,string>>> data = { { {"id", "1"}, {"name", "Alice"}, {"age", "23"}, {"score", "88"} } };
```
Comment (SamuelPalmhager) [inline tests/cpp/test_stats.cc:110]: Could use auto (?), makes code look cleaner.

**Review by RaphaelRevivor — CHANGES_REQUESTED** [review]
